packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1745,11 +1837,13 @@ class CarouselController extends ScrollController {
     }
 
     final bool hasFlexWeights = _carouselState!._flexWeights?.isNotEmpty ?? false;
-    if (_carouselState!.widget.itemBuilder != null) {
-      final int? itemCount = _carouselState!.widget.itemCount;
-      index = itemCount != null ? index.clamp(0, itemCount - 1) : 0;
-    } else {
-      index = index.clamp(0, _carouselState!.widget.children.length - 1);
+    if (!_carouselState!.widget.infinite) {
+      if (_carouselState!.widget.itemBuilder != null) {
```

Comment:
The original `if` `else` is to prevent the index to be out of range. Don't we need to constrain the index when infinite is true?

Reply:
for infinite, its already coming in constrain, it will be unnecessary operation, but I can remove the condition.